File: DXDAL/Shipping.vb
Imports System
Imports DevExpress.Xpo
Imports DevExpress.Xpo.Helpers
Imports DevExpress.Data.Filtering

Namespace SPGData
    <Persistent("tblShipMain")> _
    Public Class Shipping
        Inherits XPLiteObject
        Dim fShipMainID As Integer
        <Key()> _
        Public Property ShipMainID() As Integer
            Get
                Return fShipMainID
            End Get
            Set(ByVal value As Integer)
                SetPropertyValue(Of Integer)("ShipMainID", fShipMainID, value)
            End Set
        End Property
        Dim fShipMainDate As DateTime
        Public Property ShipMainDate() As DateTime
            Get
                Return fShipMainDate
            End Get
            Set(ByVal value As DateTime)
                SetPropertyValue(Of DateTime)("ShipMainDate", fShipMainDate, value)
            End Set
        End Property
        Dim fShipMainCarrierID As Carriers
        Public Property ShipMainCarrierID() As Carriers
            Get
                Return fShipMainCarrierID
            End Get
            Set(ByVal value As Carriers)
                SetPropertyValue(Of Carriers)("ShipMainCarrierID", fShipMainCarrierID, value)
            End Set
        End Property
        Dim fShipMainCustID As Customers
        Public Property ShipMainCustID() As Customers
            Get
                Return fShipMainCustID
            End Get
            Set(ByVal value As Customers)
                SetPropertyValue(Of Customers)("ShipMainCustID", fShipMainCustID, value)
            End Set
        End Property
        Dim fShipMainBOL As Integer
        Public Property ShipMainBOL() As Integer
            Get
                Return fShipMainBOL
            End Get
            Set(ByVal value As Integer)
                SetPropertyValue(Of Integer)("ShipMainBOL", fShipMainBOL, value)
            End Set
        End Property
        Dim fstrEnteredBy As String
        <Size(50)> _
        Public Property strEnteredBy() As String
            Get
                Return fstrEnteredBy
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("strEnteredBy", fstrEnteredBy, value)
            End Set
        End Property
        Dim fdtmEnteredOn As DateTime
        Public Property dtmEnteredOn() As DateTime
            Get
                Return fdtmEnteredOn
            End Get
            Set(ByVal value As DateTime)
                SetPropertyValue(Of DateTime)("dtmEnteredOn", fdtmEnteredOn, value)
            End Set
        End Property
        Dim fintDestination As ShippingAddresses
        Public Property intDestination() As ShippingAddresses
            Get
                Return fintDestination
            End Get
            Set(ByVal value As ShippingAddresses)
                SetPropertyValue(Of ShippingAddresses)("intDestination", fintDestination, value)
            End Set
        End Property
        Dim fstrPO As String
        <Size(50)> _
        Public Property strPO() As String
            Get
                Return fstrPO
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("strPO", fstrPO, value)
            End Set
        End Property
        Dim fstrTrailer As String
        <Size(50)> _
        Public Property strTrailer() As String
            Get
                Return fstrTrailer
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("strTrailer", fstrTrailer, value)
            End Set
        End Property
        Dim fsngSkits As Single
        Public Property sngSkits() As Single
            Get
                Return fsngSkits
            End Get
            Set(ByVal value As Single)
                SetPropertyValue(Of Single)("sngSkits", fsngSkits, value)
            End Set
        End Property
        Dim fsngTotalPallets As Single
        Public Property sngTotalPallets() As Single
            Get
                Return fsngTotalPallets
            End Get
            Set(ByVal value As Single)
                SetPropertyValue(Of Single)("sngTotalPallets", fsngTotalPallets, value)
            End Set
        End Property
        Dim fstrSeal As String
        <Size(50)> _
        Public Property strSeal() As String
            Get
                Return fstrSeal
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("strSeal", fstrSeal, value)
            End Set
        End Property
        Dim fstrRoute As String
        <Size(50)> _
        Public Property strRoute() As String
            Get
                Return fstrRoute
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("strRoute", fstrRoute, value)
            End Set
        End Property
        Dim fstrDeliveryNote As String
        Public Property strDeliveryNote() As String
            Get
                Return fstrDeliveryNote
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("strDeliveryNote", fstrDeliveryNote, value)
            End Set
        End Property
        Dim fDeliveryNoteNumber As String
        <Size(50)> _
        Public Property DeliveryNoteNumber() As String
            Get
                Return fDeliveryNoteNumber
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("DeliveryNoteNumber", fDeliveryNoteNumber, value)
            End Set
        End Property
        Dim fTotalGrossWeight As Double
        Public Property TotalGrossWeight() As Double
            Get
                Return fTotalGrossWeight
            End Get
            Set(ByVal value As Double)
                SetPropertyValue(Of Double)("TotalGrossWeight", fTotalGrossWeight, value)
            End Set
        End Property
        Dim fShipper As Boolean
        Public Property Shipper() As Boolean
            Get
                Return fShipper
            End Get
            Set(ByVal value As Boolean)
                SetPropertyValue(Of Boolean)("Shipper", fShipper, value)
            End Set
        End Property
        Dim fComments As String
        <Size(255)> _
        Public Property Comments() As String
            Get
                Return fComments
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Comments", fComments, value)
            End Set
        End Property
        Dim fShift As Shifts
        Public Property Shift() As Shifts
            Get
                Return fShift
            End Get
            Set(ByVal value As Shifts)
                SetPropertyValue(Of Shifts)("Shift", fShift, value)
            End Set
        End Property
        Dim fFreight As Boolean
        Public Property Freight() As Boolean
            Get
                Return fFreight
            End Get
            Set(ByVal value As Boolean)
                SetPropertyValue(Of Boolean)("Freight", fFreight, value)
            End Set
        End Property
        Dim fLoadedBy As Employees
        <Association("Employee-Loaded")>
        Public Property LoadedBy() As Employees
            Get
                Return fLoadedBy
            End Get
            Set(value As Employees)
                SetPropertyValue(Of Employees)("LoadedBy", fLoadedBy, value)
            End Set
        End Property
        Dim fCheckedBy As Employees
        <Association("Employee-CheckedShipping")>
        Public Property CheckedBy() As Employees
            Get
                Return fCheckedBy
            End Get
            Set(value As Employees)
                SetPropertyValue(Of Employees)("CheckedBy", fCheckedBy, value)
            End Set
        End Property
        Dim fPhysicalCondition As ShipmentConditions
        <Association("ShipingPhysicalCondition")>
        Public Property PhysicalCondition As ShipmentConditions
            Get
                Return fPhysicalCondition
            End Get
            Set(value As ShipmentConditions)
                SetPropertyValue(Of ShipmentConditions)("PhysicalCondition", fPhysicalCondition, value)
            End Set
        End Property
        Dim fLoadCondition As ShipmentConditions
        <Association("ShipingLoadCondition")>
        Public Property LoadCondition As ShipmentConditions
            Get
                Return fLoadCondition
            End Get
            Set(value As ShipmentConditions)
                SetPropertyValue(Of ShipmentConditions)("LoadCondition", fLoadCondition, value)
            End Set
        End Property
        Dim fStartTime As DateTime
        Public Property StartTime As DateTime
            Get
                Return fStartTime
            End Get
            Set(value As DateTime)
                SetPropertyValue(Of DateTime)("StartTime", fStartTime, value)
            End Set
        End Property
        Dim fFinishTime As DateTime
        Public Property FinishTime As DateTime
            Get
                Return fFinishTime
            End Get
            Set(value As DateTime)
                SetPropertyValue(Of DateTime)("FinishTime", fFinishTime, value)
            End Set
        End Property
        Dim fTemparature As Double
        Public Property Temparature As Double
            Get
                Return fTemparature
            End Get
            Set(value As Double)
                SetPropertyValue(Of Double)("Temparature", fTemparature, value)
            End Set
        End Property
        Dim fWheelsChocked As ThreeWayYesNo
        Public Property WheelsChocked As ThreeWayYesNo
            Get
                Return fWheelsChocked
            End Get
            Set(value As ThreeWayYesNo)
                SetPropertyValue(Of ThreeWayYesNo)("WheelsChocked", fWheelsChocked, value)
            End Set
        End Property
        Dim fForeignSubstance As ThreeWayYesNo
        Public Property ForeignSubstance As ThreeWayYesNo
            Get
                Return fForeignSubstance
            End Get
            Set(value As ThreeWayYesNo)
                SetPropertyValue(Of ThreeWayYesNo)("ForeignSubstance", fForeignSubstance, value)
            End Set
        End Property
        Dim fInsectActivity As ThreeWayYesNo
        Public Property InsectActivity As ThreeWayYesNo
            Get
                Return fInsectActivity
            End Get
            Set(value As ThreeWayYesNo)
                SetPropertyValue(Of ThreeWayYesNo)("InsectActivity", fInsectActivity, value)
            End Set
        End Property
        Dim fCorrectPallets As ThreeWayYesNo
        Public Property CorrectPallets As ThreeWayYesNo
            Get
                Return fCorrectPallets
            End Get
            Set(value As ThreeWayYesNo)
                SetPropertyValue(Of ThreeWayYesNo)("CorrectPallets", fCorrectPallets, value)
            End Set
        End Property
        Dim fShippingLocation As Locations
        <Association("Location-Shippings")>
        Public Property ShippingLocation As Locations
            Get
                Return fShippingLocation
            End Get
            Set(value As Locations)
                SetPropertyValue(Of Locations)("ShippingLocation", fShippingLocation, value)
            End Set
        End Property

        <Association("Shipping-Details", GetType(ShipDet)), Aggregated()> _
        Public ReadOnly Property ShippingDetails() As XPCollection(Of ShipDet)
            Get
                Return GetCollection(Of ShipDet)("ShippingDetails")
            End Get
        End Property
        <Association("Shippiing-Returns", GetType(ShippedReturns)), Aggregated()> _
        Public ReadOnly Property ShippingReturns() As XPCollection(Of ShippedReturns)
            Get
                Return GetCollection(Of ShippedReturns)("ShippingReturns")
            End Get
        End Property
        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Public Sub New()
            MyBase.New(Session.DefaultSession)
        End Sub
        Public Overrides Sub AfterConstruction()
            MyBase.AfterConstruction()
        End Sub
        Private Shared _fields As FieldsClass
        Public Shared Shadows ReadOnly Property Fields() As FieldsClass
            Get
                If ReferenceEquals(_fields, Nothing) Then
                    _fields = New FieldsClass()
                End If
                Return _fields
            End Get
        End Property
        'Created/Updated: Joel-PC\Joel on JOEL-PC at 7/06/10 4:39 PM
        Public Shadows Class FieldsClass
            Inherits DevExpress.Xpo.XPLiteObject.FieldsClass
            Public Sub New()
                MyBase.New()
            End Sub
            Public Sub New(ByVal propertyName As String)
                MyBase.New(propertyName)
            End Sub
            Public ReadOnly Property ShipMainID() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("ShipMainID"))
                End Get
            End Property
            Public ReadOnly Property ShipMainDate() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("ShipMainDate"))
                End Get
            End Property
            Public ReadOnly Property ShipMainCarrierID() As Carriers.FieldsClass
                Get
                    Return New Carriers.FieldsClass(GetNestedName("ShipMainCarrierID"))
                End Get
            End Property
            Public ReadOnly Property ShipMainCustID() As DXDAL.SPGData.Customers.FieldsClass
                Get
                    Return New DXDAL.SPGData.Customers.FieldsClass(GetNestedName("ShipMainCustID"))
                End Get
            End Property
            Public ReadOnly Property ShipMainBOL() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("ShipMainBOL"))
                End Get
            End Property
            Public ReadOnly Property strEnteredBy() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("strEnteredBy"))
                End Get
            End Property
            Public ReadOnly Property dtmEnteredOn() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("dtmEnteredOn"))
                End Get
            End Property
            Public ReadOnly Property intDestination() As DXDAL.SPGData.ShippingAddresses.FieldsClass
                Get
                    Return New DXDAL.SPGData.ShippingAddresses.FieldsClass(GetNestedName("intDestination"))
                End Get
            End Property
            Public ReadOnly Property strPO() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("strPO"))
                End Get
            End Property
            Public ReadOnly Property strTrailer() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("strTrailer"))
                End Get
            End Property
            Public ReadOnly Property sngSkits() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("sngSkits"))
                End Get
            End Property
            Public ReadOnly Property sngTotalPallets() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("sngTotalPallets"))
                End Get
            End Property
            Public ReadOnly Property strSeal() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("strSeal"))
                End Get
            End Property
            Public ReadOnly Property strRoute() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("strRoute"))
                End Get
            End Property
            Public ReadOnly Property strDeliveryNote() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("strDeliveryNote"))
                End Get
            End Property
            Public ReadOnly Property DeliveryNoteNumber() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("DeliveryNoteNumber"))
                End Get
            End Property
            Public ReadOnly Property TotalGrossWeight() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("TotalGrossWeight"))
                End Get
            End Property
            Public ReadOnly Property Shipper() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("Shipper"))
                End Get
            End Property
            Public ReadOnly Property Comments() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("Comments"))
                End Get
            End Property
            Public ReadOnly Property Shift() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("Shift"))
                End Get
            End Property
            Public ReadOnly Property Freight() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("Freight"))
                End Get
            End Property
            Public ReadOnly Property ShippingDetails() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("ShippingDetails"))
                End Get
            End Property
            Public ReadOnly Property ShippingReturns() As DevExpress.Data.Filtering.OperandProperty
                Get
                    Return New DevExpress.Data.Filtering.OperandProperty(GetNestedName("ShippingReturns"))
                End Get
            End Property
            Public ReadOnly Property LoadedBy() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("LoadedBy"))
                End Get
            End Property
            Public ReadOnly Property CheckedBy() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("CheckedBy"))
                End Get
            End Property
            Public ReadOnly Property PhysicalCondition As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("PhysicalCondition"))
                End Get
            End Property
            Public ReadOnly Property LoadCondition As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("LoadCondition"))
                End Get
            End Property
            Public ReadOnly Property StartTime As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("StartTime"))
                End Get
            End Property
            Public ReadOnly Property FinishTime As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("FinishTime"))
                End Get
            End Property
            Public ReadOnly Property Temparature As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("Temparature"))
                End Get
            End Property
            Public ReadOnly Property WheelsChocked As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("WheelsChocked"))
                End Get
            End Property
            Public ReadOnly Property ForeignSubstance As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("ForeignSubstance"))
                End Get
            End Property
            Public ReadOnly Property InsectActivity As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("InsectActivity"))
                End Get
            End Property
            Public ReadOnly Property CorrectPallets As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("CorrectPallets"))
                End Get
            End Property
            Public ReadOnly Property ShippingLocation As Locations.FieldsClass
                Get
                    Return New Locations.FieldsClass(GetNestedName("ShippingLocation"))
                End Get
            End Property

        End Class
        
    End Class
End Namespace

File: DXDAL/Shipping.cs
using System;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;

namespace DXDAL
{
    namespace SPGData
    {
        [Persistent("tblShipMain")]
        public class Shipping : XPLiteObject
        {
            private int fShipMainID;
            [Key()]
            public int ShipMainID
            {
                get => fShipMainID;
                set => SetPropertyValue<int>("ShipMainID", ref fShipMainID, value);
            }
            private DateTime fShipMainDate;
            public DateTime ShipMainDate
            {
                get => fShipMainDate;
                set => SetPropertyValue<DateTime>("ShipMainDate", ref fShipMainDate, value);
            }
            private Carriers fShipMainCarrierID;
            public Carriers ShipMainCarrierID
            {
                get => fShipMainCarrierID;
                set => SetPropertyValue("ShipMainCarrierID", ref fShipMainCarrierID, value);
            }
            private Customers fShipMainCustID;
            public Customers ShipMainCustID
            {
                get => fShipMainCustID;
                set => SetPropertyValue("ShipMainCustID", ref fShipMainCustID, value);
            }
            private int fShipMainBOL;
            public int ShipMainBOL
            {
                get => fShipMainBOL;
                set => SetPropertyValue<int>("ShipMainBOL", ref fShipMainBOL, value);
            }
            private string fstrEnteredBy;
            [Size(50)]
            public string strEnteredBy
            {
                get => fstrEnteredBy;
                set => SetPropertyValue("strEnteredBy", ref fstrEnteredBy, value);
            }
            private DateTime fdtmEnteredOn;
            public DateTime dtmEnteredOn
            {
                get => fdtmEnteredOn;
                set => SetPropertyValue<DateTime>("dtmEnteredOn", ref fdtmEnteredOn, value);
            }
            private ShippingAddresses fintDestination;
            public ShippingAddresses intDestination
            {
                get => fintDestination;
                set => SetPropertyValue("intDestination", ref fintDestination, value);
            }
            private string fstrPO;
            [Size(50)]
            public string strPO
            {
                get => fstrPO;
                set => SetPropertyValue("strPO", ref fstrPO, value);
            }
            private string fstrTrailer;
            [Size(50)]
            public string strTrailer
            {
                get => fstrTrailer;
                set => SetPropertyValue("strTrailer", ref fstrTrailer, value);
            }
            private float fsngSkits;
            public float sngSkits
            {
                get => fsngSkits;
                set => SetPropertyValue("sngSkits", ref fsngSkits, value);
            }
            private float fsngTotalPallets;
            public float sngTotalPallets
            {
                get => fsngTotalPallets;
                set => SetPropertyValue("sngTotalPallets", ref fsngTotalPallets, value);
            }
            private string fstrSeal;
            [Size(50)]
            public string strSeal
            {
                get => fstrSeal;
                set => SetPropertyValue("strSeal", ref fstrSeal, value);
            }
            private string fstrRoute;
            [Size(50)]
            public string strRoute
            {
                get => fstrRoute;
                set => SetPropertyValue("strRoute", ref fstrRoute, value);
            }
            private string fstrDeliveryNote;
            public string strDeliveryNote
            {
                get => fstrDeliveryNote;
                set => SetPropertyValue("strDeliveryNote", ref fstrDeliveryNote, value);
            }
            private string fDeliveryNoteNumber;
            [Size(50)]
            public string DeliveryNoteNumber
            {
                get => fDeliveryNoteNumber;
                set => SetPropertyValue("DeliveryNoteNumber", ref fDeliveryNoteNumber, value);
            }
            private double fTotalGrossWeight;
            public double TotalGrossWeight
            {
                get => fTotalGrossWeight;
                set => SetPropertyValue("TotalGrossWeight", ref fTotalGrossWeight, value);
            }
            private bool fShipper;
            public bool Shipper
            {
                get => fShipper;
                set => SetPropertyValue("Shipper", ref fShipper, value);
            }
            private string fComments;
            [Size(255)]
            public string Comments
            {
                get => fComments;
                set => SetPropertyValue("Comments", ref fComments, value);
            }
            private Shifts fShift;
            public Shifts Shift
            {
                get => fShift;
                set => SetPropertyValue("Shift", ref fShift, value);
            }
            private bool fFreight;
            public bool Freight
            {
                get => fFreight;
                set => SetPropertyValue("Freight", ref fFreight, value);
            }
            private Employees fLoadedBy;
            [Association("Employee-Loaded")]
            public Employees LoadedBy
            {
                get => fLoadedBy;
                set => SetPropertyValue("LoadedBy", ref fLoadedBy, value);
            }
            private Employees fCheckedBy;
            [Association("Employee-CheckedShipping")]
            public Employees CheckedBy
            {
                get => fCheckedBy;
                set => SetPropertyValue("CheckedBy", ref fCheckedBy, value);
            }
            private ShipmentConditions fPhysicalCondition;
            [Association("ShipingPhysicalCondition")]
            public ShipmentConditions PhysicalCondition
            {
                get => fPhysicalCondition;
                set => SetPropertyValue("PhysicalCondition", ref fPhysicalCondition, value);
            }
            private ShipmentConditions fLoadCondition;
            [Association("ShipingLoadCondition")]
            public ShipmentConditions LoadCondition
            {
                get => fLoadCondition;
                set => SetPropertyValue("LoadCondition", ref fLoadCondition, value);
            }
            private DateTime fStartTime;
            public DateTime StartTime
            {
                get => fStartTime;
                set => SetPropertyValue<DateTime>("StartTime", ref fStartTime, value);
            }
            private DateTime fFinishTime;
            public DateTime FinishTime
            {
                get => fFinishTime;
                set => SetPropertyValue<DateTime>("FinishTime", ref fFinishTime, value);
            }
            private double? fTemparature;
            public double? Temparature
            {
                get => fTemparature;
                set => SetPropertyValue("Temparature", ref fTemparature, value);
            }
            private ThreeWayYesNo? fWheelsChocked;
            public ThreeWayYesNo? WheelsChocked
            {
                get => fWheelsChocked;
                set => SetPropertyValue("WheelsChocked", ref fWheelsChocked, value);
            }
            private ThreeWayYesNo? fForeignSubstance;
            public ThreeWayYesNo? ForeignSubstance
            {
                get => fForeignSubstance;
                set => SetPropertyValue("ForeignSubstance", ref fForeignSubstance, value);
            }
            private ThreeWayYesNo? fInsectActivity;
            public ThreeWayYesNo? InsectActivity
            {
                get => fInsectActivity;
                set => SetPropertyValue("InsectActivity", ref fInsectActivity, value);
            }
            private ThreeWayYesNo? fCorrectPallets;
            public ThreeWayYesNo? CorrectPallets
            {
                get => fCorrectPallets;
                set => SetPropertyValue("CorrectPallets", ref fCorrectPallets, value);
            }
            private Locations fShippingLocation;
            [Association("Location-Shippings")]
            public Locations ShippingLocation
            {
                get => fShippingLocation;
                set => SetPropertyValue("ShippingLocation", ref fShippingLocation, value);
            }

            [Association("Shipping-Details", typeof(ShipDet))]
            [Aggregated()]
            public XPCollection<ShipDet> ShippingDetails => GetCollection<ShipDet>("ShippingDetails");
            [Association("Shippiing-Returns", typeof(ShippedReturns))]
            [Aggregated()]
            public XPCollection<ShippedReturns> ShippingReturns => GetCollection<ShippedReturns>("ShippingReturns");
            public Shipping(Session session) : base(session)
            {
            }
            public Shipping() : base(Session.DefaultSession)
            {
            }
            public override void AfterConstruction()
            {
                base.AfterConstruction();
            }
            private static FieldsClass _fields;
            public static new FieldsClass Fields
            {
                get
                {
                    if (ReferenceEquals(_fields, null))
                        _fields = new FieldsClass();
                    return _fields;
                }
            }
            // Created/Updated: Joel-PC\Joel on JOEL-PC at 7/06/10 4:39 PM
            public new class FieldsClass : PersistentBase.FieldsClass
            {
                public FieldsClass() : base()
                {
                }
                public FieldsClass(string propertyName) : base(propertyName)
                {
                }
                public OperandProperty ShipMainID => new OperandProperty(GetNestedName("ShipMainID"));
                public OperandProperty ShipMainDate => new OperandProperty(GetNestedName("ShipMainDate"));
                public Carriers.FieldsClass ShipMainCarrierID => new Carriers.FieldsClass(GetNestedName("ShipMainCarrierID"));
                public Customers.FieldsClass ShipMainCustID => new Customers.FieldsClass(GetNestedName("ShipMainCustID"));
                public OperandProperty ShipMainBOL => new OperandProperty(GetNestedName("ShipMainBOL"));
                public OperandProperty strEnteredBy => new OperandProperty(GetNestedName("strEnteredBy"));
                public OperandProperty dtmEnteredOn => new OperandProperty(GetNestedName("dtmEnteredOn"));
                public ShippingAddresses.FieldsClass intDestination => new ShippingAddresses.FieldsClass(GetNestedName("intDestination"));
                public OperandProperty strPO => new OperandProperty(GetNestedName("strPO"));
                public OperandProperty strTrailer => new OperandProperty(GetNestedName("strTrailer"));
                public OperandProperty sngSkits => new OperandProperty(GetNestedName("sngSkits"));
                public OperandProperty sngTotalPallets => new OperandProperty(GetNestedName("sngTotalPallets"));
                public OperandProperty strSeal => new OperandProperty(GetNestedName("strSeal"));
                public OperandProperty strRoute => new OperandProperty(GetNestedName("strRoute"));
                public OperandProperty strDeliveryNote => new OperandProperty(GetNestedName("strDeliveryNote"));
                public OperandProperty DeliveryNoteNumber => new OperandProperty(GetNestedName("DeliveryNoteNumber"));
                public OperandProperty TotalGrossWeight => new OperandProperty(GetNestedName("TotalGrossWeight"));
                public OperandProperty Shipper => new OperandProperty(GetNestedName("Shipper"));
                public OperandProperty Comments => new OperandProperty(GetNestedName("Comments"));
                public OperandProperty Shift => new OperandProperty(GetNestedName("Shift"));
                public OperandProperty Freight => new OperandProperty(GetNestedName("Freight"));
                public OperandProperty ShippingDetails => new OperandProperty(GetNestedName("ShippingDetails"));
                public OperandProperty ShippingReturns => new OperandProperty(GetNestedName("ShippingReturns"));
                public OperandProperty LoadedBy => new OperandProperty(GetNestedName("LoadedBy"));
                public OperandProperty CheckedBy => new OperandProperty(GetNestedName("CheckedBy"));
                public OperandProperty PhysicalCondition => new OperandProperty(GetNestedName("PhysicalCondition"));
                public OperandProperty LoadCondition => new OperandProperty(GetNestedName("LoadCondition"));
                public OperandProperty StartTime => new OperandProperty(GetNestedName("StartTime"));
                public OperandProperty FinishTime => new OperandProperty(GetNestedName("FinishTime"));
                public OperandProperty Temparature => new OperandProperty(GetNestedName("Temparature"));
                public OperandProperty WheelsChocked => new OperandProperty(GetNestedName("WheelsChocked"));
                public OperandProperty ForeignSubstance => new OperandProperty(GetNestedName("ForeignSubstance"));
                public OperandProperty InsectActivity => new OperandProperty(GetNestedName("InsectActivity"));
                public OperandProperty CorrectPallets => new OperandProperty(GetNestedName("CorrectPallets"));
                public Locations.FieldsClass ShippingLocation => new Locations.FieldsClass(GetNestedName("ShippingLocation"));
            }
        }
    }
}
